server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryParseResponse.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportResponse;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: parse response from extensions
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryParseResponse extends TransportResponse {
```

Comment:
Let's not create a separate class for every boolean response. We already have `IndicesModuleNameResponse` doing something similar. We can make it more generic and use it for IndicesModuleNameResponse and NamedWriteableRegistryParseResponse

Reply:
`BooleanResponse<NamedWritableRegistry>` and `BooleanResponse<IndicesModuleName>`?